jbehave-core/src/main/java/org/jbehave/core/condition/DefaultConditionChecker.java

```diff
@@ -0,0 +1,21 @@
+package org.jbehave.core.condition;
+
+import java.lang.reflect.Constructor;
+import java.util.function.Predicate;
+
+public class DefaultConditionChecker implements ConditionChecker {
```

Comment:
```suggestion
public class ReflectionBasedStepConditionMatcher implements StepConditionMatcher {
```

Reply:
Does it make sense to add the implementation of `InstanceStepConditionMatcher` that will contain instances of all available conditions?